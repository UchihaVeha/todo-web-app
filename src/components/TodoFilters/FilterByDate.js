// @flow
import * as React from 'react';
import { withStyles } from 'material-ui/styles';
import DatePicker from 'material-ui-before/DatePicker';
import { IconButton, Typography } from 'material-ui';
import BeforeIcon from 'material-ui-icons/ArrowBack';
import NextIcon from 'material-ui-icons/ArrowForward';
import keycode from 'keycode';

declare var Intl: any;

const styles = theme => ({
  day: {
    textAlign: 'center',
    display: 'flex',
    justifyContent: 'space-between',
    color: theme.palette.common.darkWhite
  },
  pointed: {
    cursor: 'pointer'
  },
  datePicker: {
    visibility: 'hidden'
  },
  button: {
    alignSelf: 'flex-end'
  }
});

type Props = {
  value: string,
  onChange: string => void,
  classes: Return<typeof styles>
};

class FilterByDate extends React.Component<Props> {
  picker: ?Object;

  handleChange = (e: null, value: Date) => {
    this.props.onChange(value.toDateString());
  };

  handleNextDay = () => {
    const date = new Date(this.props.value);
    date.setDate(date.getDate() + 1);
    this.props.onChange(date.toDateString());
  };

  handlePrevDay = () => {
    const date = new Date(this.props.value);
    date.setDate(date.getDate() - 1);
    this.props.onChange(date.toDateString());
  };

  handleEnterOnDay = (event: SyntheticKeyboardEvent<>) => {
    if (keycode(event) === 'enter' && this.picker) {
      this.picker.handleTouchTap();
    }
  };

  handleClickOnDay = () => {
    if (this.picker) this.picker.handleTouchTap();
  };

  render() {
    const { value, classes } = this.props;
    const date = new Date(value);
    const day = date.toLocaleString('en', {
      day: 'numeric'
    });
    const month = date.toLocaleString('en', {
      month: 'long'
    });
    const year = date.toLocaleString('en', {
      year: 'numeric'
    });
    return (
      <div>
        <div className={classes.day}>
          <IconButton
            color="inherit"
            aria-label="Menu"
            className={classes.button}
            onClick={() => this.handlePrevDay()}
          >
            <BeforeIcon />
          </IconButton>
          <div
            role="button"
            tabIndex={0}
            className={classes.pointed}
            onKeyPress={this.handleEnterOnDay}
            onClick={() => this.handleClickOnDay()}
          >
            <Typography type="display4" color="inherit">
              {day}
            </Typography>
          </div>
          <IconButton
            color="inherit"
            aria-label="Menu"
            className={classes.button}
            onClick={() => this.handleNextDay()}
          >
            <NextIcon />
          </IconButton>
        </div>
        <div
          role="button"
          tabIndex={0}
          className={classes.pointed}
          onKeyPress={this.handleEnterOnDay}
          onClick={() => this.handleClickOnDay()}
        >
          <Typography type="title" align="center" color="secondary">
            {month} {year}
          </Typography>
        </div>

        <DatePicker
          ref={ref => {
            this.picker = ref;
          }}
          id="datepicker"
          className={classes.datePicker}
          value={new Date(value)}
          onChange={this.handleChange}
          autoOk
        />
      </div>
    );
  }
}

export default withStyles(styles, { name: 'FilterByDate' })(FilterByDate);
