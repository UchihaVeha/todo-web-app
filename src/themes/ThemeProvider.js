import React from 'react';
import PropTypes from 'prop-types';

import MuiThemeProviderBefore from 'material-ui-before/styles/MuiThemeProvider';
import { MuiThemeProvider, createMuiTheme } from 'material-ui/styles';
import { cyan, yellow } from 'material-ui/colors';

const theme = createMuiTheme({
  palette: {
    primary: cyan,
    secondary: {
      ...yellow,
      A200: yellow['400']
    }
  }
});

const AppThemeProvider = ({ children }) => (
  <MuiThemeProvider theme={theme}>
    <MuiThemeProviderBefore>{children}</MuiThemeProviderBefore>
  </MuiThemeProvider>
);

AppThemeProvider.propTypes = {
  children: PropTypes.node.isRequired
};

export default AppThemeProvider;
