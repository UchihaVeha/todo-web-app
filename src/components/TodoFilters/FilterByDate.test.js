import React from 'react';
import { shallow, mount } from 'enzyme';
import { IconButton } from 'material-ui';
import ThemeProvider from 'themes/ThemeProvider';
import FilterByDate from './FilterByDate';

describe('TodoFilters/FilterByDate component', () => {
  let wrapper;
  const date = '2017-10-28T21:00:00.000Z';
  const prevDay = 'Sat Oct 28 2017';
  const nextDay = 'Mon Oct 30 2017';

  it('should render withStyles', () => {
    wrapper = shallow(<FilterByDate value={date} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component', () => {
    wrapper = shallow(<FilterByDate value={date} />);
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should handle next day', () => {
    const onChange = jest.fn();
    wrapper = mount(
      <ThemeProvider>
        <FilterByDate value={date} onChange={onChange} />
      </ThemeProvider>
    );
    wrapper
      .find(IconButton)
      .at(1)
      .simulate('click', {});
    expect(onChange).toHaveBeenCalledWith(nextDay);
  });

  it('should handle prev day', () => {
    const onChange = jest.fn();
    wrapper = mount(
      <ThemeProvider>
        <FilterByDate value={date} onChange={onChange} />
      </ThemeProvider>
    );
    wrapper
      .find(IconButton)
      .at(0)
      .simulate('click', {});
    expect(onChange).toHaveBeenCalledWith(prevDay);
  });
});
