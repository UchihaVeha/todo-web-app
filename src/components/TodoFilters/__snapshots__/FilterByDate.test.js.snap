// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFilters/FilterByDate component should render component 1`] = `
<div>
  <div
    className="FilterByDate-day-1"
  >
    <withStyles(IconButton)
      aria-label="Menu"
      className="FilterByDate-button-4"
      color="inherit"
      onClick={[Function]}
    >
      <pure(ArrowBack) />
    </withStyles(IconButton)>
    <div
      className="FilterByDate-pointed-2"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <withStyles(Typography)
        color="inherit"
        type="display4"
      >
        29
      </withStyles(Typography)>
    </div>
    <withStyles(IconButton)
      aria-label="Menu"
      className="FilterByDate-button-4"
      color="inherit"
      onClick={[Function]}
    >
      <pure(ArrowForward) />
    </withStyles(IconButton)>
  </div>
  <div
    className="FilterByDate-pointed-2"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <withStyles(Typography)
      align="center"
      color="secondary"
      type="title"
    >
      October
       
      2017
    </withStyles(Typography)>
  </div>
  <DatePicker
    autoOk={true}
    className="FilterByDate-datePicker-3"
    container="dialog"
    disableYearSelection={false}
    disabled={false}
    firstDayOfWeek={1}
    hideCalendarDate={false}
    id="datepicker"
    onChange={[Function]}
    openToYearSelection={false}
    style={Object {}}
    value={2017-10-28T21:00:00.000Z}
  />
</div>
`;

exports[`TodoFilters/FilterByDate component should render withStyles 1`] = `
<FilterByDate
  classes={
    Object {
      "button": "FilterByDate-button-4",
      "datePicker": "FilterByDate-datePicker-3",
      "day": "FilterByDate-day-1",
      "pointed": "FilterByDate-pointed-2",
    }
  }
  value="2017-10-28T21:00:00.000Z"
/>
`;
